mlflow/pyfunc/__init__.py

```diff
@@ -1112,25 +1229,12 @@ def _predict_row_batch(predict_fn, args):
                     field_values = field_values.astype(np_type)
 
                 elif type(field_type) == ArrayType:
-                    elem_type = field_type.elementType
-                    if type(elem_type) not in spark_primitive_type_to_np_type:
-                        raise MlflowException(
-                            "Unsupported array type field with element type "
-                            f"{elem_type.simpleString()} in struct type.",
-                            error_code=INVALID_PARAMETER_VALUE,
-                        )
-                    np_type = spark_primitive_type_to_np_type[type(elem_type)]
-
-                    field_values = [
-                        np.array(v, dtype=np.dtype(type(elem_type))) for v in field_values
-                    ]
-
+                    field_values = _convert_array_values(field_values, field_type)
```

Comment:
This call is missing an arg, right? `spark_primitive_type_to_np_type` is undefined